src/marketaggregator.ts

```diff
@@ -103,16 +101,23 @@ export class MarketAggregator {
       ...coingeckoMarketDataMap,
     };
 
-    const stepMarketData = coingeckoMarketDataMap[STEP_MINT];
-    if (stepMarketData) {
-      const xStepDataMap = await this.xStep.query(stepMarketData.price);
-      markets = { ...markets, ...xStepDataMap };
+    for await (const source of this.stakedSources) {
```

Comment:
I could be wrong, but `for await` might not be needed since already in async function. I think a standard `for` loop works. Ed caught this me doing this too on one of my PR's 😅